src/transaction/transaction_manager.cpp

```diff
@@ -33,6 +33,21 @@ TransactionContext *TransactionManager::BeginTransaction() {
 
   // Do the allocation outside of any critical section
   auto *const result = new TransactionContext(start_time, start_time + INT64_MIN, buffer_pool_, log_manager_, this);
+  return result;
+}
+
+TransactionContext *TransactionManager::BeginTransaction(timestamp_t timestamp) {
```

Comment:
We probably need to do an in-depth analysis of whether fudged timestamps invalidates the assumptions inherent for how we solved the three-way race using the gate and latch.

Reply:
Removed function